tests/playwright/sanity/container-grid.test.js

```diff
@@ -119,6 +119,122 @@ test.describe( 'Container tests', () => {
+			await expect( gridOutline ).toBeVisible();
+		} );
+
+		await test.step( 'Assert number of child elements when there are less Items than grid cells', async () => {
+			// Arrange.
+			gridOutlineChildrenInitialValue = 12;
+
+			// Act.
+			await editor.setSliderControlValue( 'grid_columns_grid', '4' );
+			await editor.setSliderControlValue( 'grid_rows_grid', '3' );
+
+			// Assert.
+			await expect( gridOutlineChildren ).toHaveCount( gridOutlineChildrenInitialValue );
+		} );
+
+		await test.step( 'Assert outline match responsive grid', async () => {
+			gridOutlineChildrenInitialValue = 4;
+
+			// Open responsive bar and select mobile view
+			const firstGridOutlineChildren = gridOutline.first().locator( '.e-grid-outline-item' );
+			await editor.changeResponsiveView( 'tablet' );
+			await editor.setSliderControlValue( 'grid_columns_grid_tablet', '2' );
+			await editor.setSliderControlValue( 'grid_rows_grid_tablet', '2' );
+
+			await expect( gridOutline.first() ).toBeVisible();
+
+			await editor.setSelectControlValue( 'content_width', 'full' );
+
+			await expect( firstGridOutlineChildren ).toHaveCount( gridOutlineChildrenInitialValue );
+
+			await editor.removeElement( parentContainer );
+			await editor.changeResponsiveView( 'desktop' );
+		} );
+
+		await test.step( 'Assert number of child elements when there are more Items than grid cells', async () => {
+			// Arrange
+			const newParentContainer = await editor.addElement( { elType: 'container' }, 'document' ),
+				numberOfWidgets = 5,
+				expectedNumberOfChildren = 6;
+
+			await editor.setSelectControlValue( 'container_type', 'grid' );
+
+			await editor.setSliderControlValue( 'grid_columns_grid', '2' );
+			await editor.setSliderControlValue( 'grid_rows_grid', '2' );
+
+			// Act.
+			for ( let i = 0; i < numberOfWidgets; i++ ) {
+				await editor.addWidget( 'button', newParentContainer );
+			}
+
+			// Assert.
+			await expect( gridOutlineChildren ).toHaveCount( expectedNumberOfChildren );
+		} );
+	} );
+
+	test( 'Test grid outline multiple controls', async ( { page }, testInfo ) => {
+		const wpAdmin = new WpAdminPage( page, testInfo );
+		const editor = await wpAdmin.useElementorCleanPost(),
+			frame = editor.getPreviewFrame(),
+			gridOutline = await frame.locator( '.e-grid-outline' );
+
+		// Arrange.
+		await test.step( 'Arrange', async () => {
+			await editor.addElement( { elType: 'container' }, 'document' );
+			await editor.closeNavigatorIfOpen();
+			await editor.setSelectControlValue( 'container_type', 'grid' );
+		} );
+
+		await test.step( 'Check that Gaps control effects the grid outline ', async () => {
+			const desiredGapValue = '30px 30px';
+			await page.locator( '.elementor-control-grid_gaps .elementor-control-gap >> nth=0' ).locator( 'input' ).fill( '30' );
+
+			await expect( gridOutline ).toHaveCSS( 'grid-gap', desiredGapValue );
+		} );
+
+		await test.step( 'Check that Custom control is set to grid outline', async () => {
```

Comment:
Maybe 'custom control'.

Reply:
The control name is Custom, I think we should leave it this way.